From e6faf33ffa866c7b4e0366d28eace58e5f64cb44 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 13 Jun 2018 19:01:39 +0800
Subject: [PATCH 4256/5242] MLK-18605-3 drm/bridge: sec-dsim: refuse unknown
 dsi device attach

commit  61d65e61d071cf26784f74f35d5e7032417dffe5 from
https://source.codeaurora.org/external/imx/linux-imx.git

When a dsi device who is neither a bridge nor a panel requests
to be attached to the host, refuse this request directly.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 2e80acc8a611327fcc77d2e73515bc062cdc4233)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index b2f0ade..f44d6c8 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -298,7 +298,12 @@ static int sec_mipi_dsim_host_attach(struct mipi_dsi_host *host,
 	if (!dsim->next) {
 		/* 'dsi' must be panel device */
 		panel = of_drm_find_panel(dsi->dev.of_node);
-		WARN_ON(!panel);
+
+		if (!panel) {
+			dev_err(dev, "refuse unknown dsi device attach\n");
+			WARN_ON(!panel);
+			return -ENODEV;
+		}
 
 		/* Don't support multiple panels */
 		if (dsim->panel && panel && dsim->panel != panel) {
-- 
1.7.9.5

